storm-client/src/jvm/org/apache/storm/utils/Utils.java

```diff
@@ -124,11 +124,47 @@
     private static String memoizedLocalHostnameString = null;
     public static final Pattern TOPOLOGY_KEY_PATTERN = Pattern.compile("^[\\w \\t\\._-]+$", Pattern.UNICODE_CHARACTER_CLASS);
 
+    public static final String NUMA_MEMORY_IN_MB = "numa.memory.mb";
+    public static final String NUMA_CORES = "numa.cores";
+    public static final String NUMA_PORTS = "numa.ports";
+    public static final String NUMA_GENERIC_RESOURCES_MAP = "numa.generic.resources.map";
+
     static {
         localConf = readStormConfig();
         serializationDelegate = getSerializationDelegate(localConf);
     }
 
+    /**
+     * Return supervisor numa configuration.
+     * @param stormConf stormConf
+     * @return getNumaMap
+     */
+    public static Map<String, Object> getNumaMap(Map<String, Object> stormConf) {
+        Object numa = stormConf.get(Config.SUPERVISOR_NUMA_META);
+        if (numa == null) {
+            return Collections.emptyMap();
+        }
+        return (Map<String, Object>) numa;
+    }
+
+    /**
+     * getNumaIdForPort.
```

Comment:
This one seems to be used only in storm-server.  Can we make this one SupervisorUtils?

Please update the java doc to indicate this is specific to a single supervisor.  stormConf could be renamed supervisorConf for clarity on this.

Reply:
https://issues.apache.org/jira/browse/STORM-3615